packages/platform-android/src/commands/runAndroid/emulator.ts

```diff
@@ -42,32 +52,31 @@ const launchEmulator = async (emulatorName: string, adbPath: string) => {
       clearInterval(bootCheckInterval);
     };
 
-    cp.on('exit', () => {
+    cp.on('exit', (isEmulatorLaunched: 0 | 1) => {
       cleanup();
-      reject('Emulator exited before boot.');
+
+      if (!isEmulatorLaunched) {
+        return reject({message: 'Emulator exited before boot.'});
+      }
+
+      return resolve(Adb.getDevices(adbPath));
```

Comment:
Again, no need to resolve here with `devices`. You could technically resolve with a `device` object of what you have just launched, or its UDID.